### DISCUSSION TITLE
AD The provided client secret keys for app are expired

### DISCUSSION THREAD

**Discussion by MuhammadNaeemAkhtar** [datalust/seq-tickets#2215] [Help]
I am currently facing an issue with the Azure Directory (AD) authentication.

Details:
Deployment Platform: Amazon ECS with EFS as persistent volume
Container Image: datalust/seq
Issue: The AD client secret keys have expired, causing authentication failures. Because of that I am not even able to login to update the secret.

Specific Questions:
Is it possible to update the secret directly in the file path (NOTE: I can execute commands inside the ECS task)?

**Comment by nblumhardt** [comment]
Hi @MuhammadNaeemAkhtar 

Yes, it's possible to do this either through a shell in the container, or directly by writing to the EFS volume, using init scripts:

https://docs.datalust.co/docs/init-scripts

The init script is used to temporarily put the container into username/password auth mode; see:

The init script contents should be:

```sh
#!/bin/bash
seqsvr auth reset -u <username> -p <new password>
```

After this has been created, restarting the container will safely run it.
